napari/utils/progress.py

```diff
@@ -0,0 +1,245 @@
+import inspect
+from contextvars import ContextVar
+from typing import Iterable, Optional
+
+from qtpy import QtCore
+from qtpy.QtWidgets import (
+    QApplication,
+    QHBoxLayout,
+    QLabel,
+    QProgressBar,
+    QVBoxLayout,
+    QWidget,
+)
+from tqdm import tqdm
+
+from .._qt.utils import get_viewer_instance
+
+IS_NESTED = ContextVar('IS_NESTED', default=False)
```

Comment:
see comments below about having this be more useful than a boolean (perhaps `id(self)` in `progress`?)

```python
CURRENT_PBAR: ContextVar[int] = ContextVar('CURRENT_PBAR', default=0)
```